### DISCUSSION TITLE
inject inputs for py-script

### DISCUSSION THREAD

**Discussion by bugzpodder** [pyscript/pyscript#1466] [Q&A] [answered]
I think py-script has an output param that lets you specify outputs.  I am curious if we can pass in some sort of objects so its automatically available in the block as some kind of injected input.  perhaps in python read some kind of id on the py-script tag

specific use case is i have a bunch of <py-script> that runs the same code with slighly different input, and perhaps instead of hard code it in python i want to specify it in js/html.  (eg each py-script tag runs the same function to draws a box with different color)

also for the upcoming webworker, how do we pass inputs from js to python?  how will the webworker have access to the parent's global object (https://docs.pyscript.net/latest/guides/passing-objects.html)?

**Comment by bugzpodder** [comment, marked as answer]
per @JeffersGlass:
you can do for v2023.03.1:
```
#demo.py
from pyscript import get_current_display_target

print(get_current_display_target._id)
```
or [current latest](https://github.com/pyscript/pyscript/blob/b0df96b13f19dbb4cc59063c10fd01577db8a96c/pyscriptjs/src/python/pyscript/_internal.py#L67):
```
from pyscript._internal import DISPLAY_TARGET

print(DISPLAY_TARGET)
```

This is a hack based on knowledge of the internals, so pinning of pyscript version is mandatory, because it's gonna get broken/changed in the next release

**Reply by JeffersGlass** [reply]
You could even take this a step further if you wanted to and use that ID to get a reference to the element itself, and start adding additional attributes there:
```html
<!--index.html-->
<py-script data-number="1" src="demo.py"></py-script>
```
```py
#demo.py, pyscript 2023.03.1
import js
from pyscript import get_current_display_target

el = js.document.getElementById(get_current_display_target._id)
print(el.getAttribute("data-number")) # prints "1"
```
